src/nodes.cpp

```diff
@@ -20,12 +20,28 @@ namespace sonata {
 
 //--------------------------------------------------------------------------------------------------
 
+namespace {
+
+const char* NODE_ID_DSET = "node_id";
+
+}  // unnamed namespace
+
+
 NodePopulation::NodePopulation(const std::string& h5FilePath,
                                const std::string& csvFilePath,
                                const std::string& name)
     : Population(h5FilePath, csvFilePath, name, ELEMENT) {}
 
 
+std::vector<NodeID> NodePopulation::nodeIDs(const Selection& selection) const {
+    HDF5_LOCK_GUARD
+    if (H5Lexists(impl_->h5Root.getId(), NODE_ID_DSET, H5P_DEFAULT) > 0) {
```

Comment:
Here's no HighFive function for this? Could also be a `try … catch …`

Reply:
Indeed there is a function in HighFive. I missed it. 
Changed.

Reply:
Thank you!